### ISSUE TITLE
Suggestion: improve documentation regarding setting variables and copyVariablesToLocalForTasks flag 

### ISSUE THREAD

**Issue by arvyy** [Activiti/Activiti#4330]
I have a following process (attached screenshot). During user task, user's input would be set as a variable with `runtimeService.setVariable(executionId, "comment", commentValue)`. During gateway branch, same variable would be fetched with `runtimeService.getVariable(executionId, "comment")` and compared against hardcoded value "not ok".

The expected behavior was:
User enters input "not ok", saves, and completes the task -> (activiti loops back to user task) -> User sees input "not ok", edits it to be "ok", saves, and completes the task -> activiti completes the process.

The observed behavior was:
User enters input "not ok", saves, and completes the task -> (activiti loops back to user task) -> User sees input "not ok", edits it to be "ok", saves, and completes the task ->  (activiti loops back to user task) -> User sees input "not ok".

In other words, the variable was seemingly stuck on initial value. After debugging through activiti internals, I found the culprit was copyVariablesToLocalForTasks setting which defaulted to true (using spring boot). It seems said flag was causing setVariable call to non-intuitively apply to task-local variable on repeat entries when it was expected to bubble to process scope.

I don't think this is a bug thus I'm not providing a reproduction. But I think this should be more clearly documented in the section about variables to watch out for the behavior caused by copyVariablesToLocalForTasks defaulting to true.
 
![image](https://github.com/Activiti/Activiti/assets/15849668/669a3dda-12c9-49b8-808b-4caed4cac552)
